data-structures/linked_list/linked_list.py

```diff
@@ -68,10 +68,22 @@ def insert_after(self, value, newVal):
                 current._next = Node(newVal, current._next)
             current = current._next
 
+    def kthFromEnd(self, key):
+        iter = self._len - (key + 1)
+        if self._len == 0:
+            return 'LinkedList is empty.'
+        if 0 > key or key > self._len:
+            return False
```

Comment:
Since the spec calls for exceptions in these cases, you should be raising an exception instead of returning things in these `if` statements.